contracts/payment/RefundEscrow.sol

```diff
@@ -0,0 +1,91 @@
+pragma solidity ^0.4.23;
+
+import "./ConditionalEscrow.sol";
+import "../ownership/Ownable.sol";
+
+
+/**
+ * @title RefundEscrow
+ * @dev Escrow that holds investor funds for a unique benefitiary, and allows for
+ * either withdrawal by the benefiatiary, or refunds to the investors.
+ */
+contract RefundEscrow is ConditionalEscrow, Ownable {
+  enum State { Active, Refunding, Closed }
+
+  event Closed();
+  event RefundsEnabled();
+  event Refunded(address indexed investor, uint256 weiAmount);
+
+  State public state;
+  address public beneficiary;
+
+  /**
+   * @dev Constructor.
+   * @param _beneficiary The beneficiary of the investments.
+   */
+  constructor(address _beneficiary) public {
+    require(_beneficiary != address(0));
+    beneficiary = _beneficiary;
+    state = State.Active;
+  }
+
+  /**
+   * @dev Stores funds that may later be refunded.
+   * @param _investor The address funds will be sent to if a refund occurs.
+   */
+  function invest(address _investor) payable public {
```

Comment:
Why did you decide to use a separate `invest` function instead of `deposit`? From what we discussed offline, I think it was because conceptually the argument to `deposit` is the payee and in this case the investor is not necessarily the payee. I'm not convinced by that. The concept of investment feels foreign to the escrow. I would stick to the base `Escrow` interface and use `deposit` with the additional state constraint.

Reply:
agreed on `deposit` being better than `invest`. Maybe `depositFor(_payee)` to make it explicit that you're not specifically depositing for yourself?

Reply:
Agree on `invest` being foreign to the escrow itself. I wanted to make the distinction clear between the base escrow, where you `deposit` a payment to someone, and the refundable, where you `deposit` a payment to the beneficiary, but specify the refund address. Maybe we should call them `refundees` instead of `investors`?

Reply:
`refundee` is better.

The error may also be in using the term `payee` in the base `Escrow`, but I can't think of a better name right now.